ingress/controllers/nginx/controller.go

```diff
@@ -797,9 +860,24 @@ func (lbc *loadBalancerController) getPemsFromIngress(data []interface{}) map[st
 		}
 	}
 
+	lbc.secrMetadataLock.Lock()
```

Comment:
I'm fine if you think it's required. I usually try to avoid locking unless I absolutely need to. If you just need a locked map, suggest using cache.Store (or cache.ThreadSafeStore as the need may be:https://github.com/kubernetes/kubernetes/blob/master/pkg/client/cache/thread_safe_store.go).